### ISSUE TITLE
MySQL Dump target should not have hardcoded path

### ISSUE THREAD

**Issue by ShaunaGordon** [phansible/phansible#224]
The code in mysql/main.yml hard codes "/vagrant/" into the dump target path:

```
mysql_db: target=/vagrant/{{ mysql.dump }}
```

This fails when the input path is absolute, or if the share root is different from "/vagrant/". This should probably be configurable and/or autodetected when generating the file.

**Comment by naxhh** [comment]
I've never used this feature.. can you tell me how do you use it?
When you say absolute path you use always te document root path or do you specify a new shared folder in the vagrantfile and then you specify that path?

**Comment by ShaunaGordon** [comment]
phansible.com provides a "sql dump" field, but doesn't give any indication whether it should be an absolute or relative path, so I put in the absolute path.

In my case, I have my host vagrant folder mapped to /var/www/project in the vagrant box, so my MySQL dump location is something like /var/www/project/myDump.sql. With the "/vagrant/" hardcoded, Ansible sees it as "/vagrant//var/www/project/myDump.sql" which obviously fails.

I can see a few of ways of dealing with this:

# Notify users on input to fill in the field with a path relative to /vagrant.
# Remove the /vagrant/ hardcoding and have the user put in an absolute or relative path (and note to where it's relative on the website)
# Auto-detect whether it's an absolute or relative path and adjust the playbook file accordingly (this is probably a little more involved, but can allow for a future feature of allowing the vm-side sync folder to be customized from within the Phansible interface if desired).

**Comment by naxhh** [comment]
hmm

I think the more flexible solution is to force the user to specify the full path (also tell him that the full path is expected)

This way you can map a new folder for the dumps or create a specific app task that copy the dumps from one place to that folder. And you are not required to keep them with your codebase, but you can if you want...

Will that fit your use case?
I'll also wait for some input from other people to see if there are more points of view from this :)

**Comment by mikeSimonson** [comment]
I also think forcing the full path is the simplest solution.

**Comment by ShaunaGordon** [comment]
Agreed. The full path was what I defaulted to when not given indication of how it should be entered, so it follows my use case.

**Comment by naxhh** [comment]
As a summary the task should be:
- Making sure the UI explains you need to set the full path.
- Create the default value to a full-path example.
- Change mysql-role to allow full-path in the dump
https://github.com/phansible/role-mysql/blob/master/tasks/main.yml#L31

**Comment by debo** [comment]
@naxhh wouldn't be actually better to inherit the path from the shared folder one?

**Comment by naxhh** [comment]
I don't normally keep the sql dumps in the same place as the code. So for me is better to give flexibility on this.
I do create a new shared path and load sql from them. But maybe the most common case will be to use the default path and allow it to change from the options.

I let the decision to you. The main priority of the task thought is to add an explanation if full or relative path should be added.

**Comment by debo** [comment]
What I meant is that at present we have two shared path, the default shared one and the document root, I generally share the code in the document root and everything else in the main shared folder. But yes I think it's all down to make clearer what's what.
